### DISCUSSION TITLE
Critique of the new floating search button

### DISCUSSION THREAD

**Discussion by dAtA-TRoN** [fast4x/RiMusic#1496] [General]
Critique of the new floating search button

First of all, a huge thank you for the new update. Once again, many great new things have been added.

But there are, of course, things that might need to be reconsidered upon closer inspection. One of these things, in my opinion, is the new floating search button. It's a great addition to RiMusic, but in its current form, I don't think it's implemented as nicely as it could be. For one, I don't understand why this button should be movable. I haven't encountered any other app where this is possible. One could argue for ergonomics, but I really question if that's necessary. And if it is, then this feature should not be as prominent as it currently is. Typically, you set something like this once and leave it be. With the current implementation, there's even a risk of accidentally changing the positioning because it's so easily accessible. Therefore, I would suggest that if this feature must remain, it should be moved to the settings. From there, users should have the option to set the button's position, which would then remain fixed during further use of the app. And once this is addressed, the activation of the settings button could be implemented by simply holding the floating button for a short time. This has several advantages in my opinion: it makes the operation calmer than performing a double tap on these small buttons. Sure, it can be done, but it doesn't feel good, as even holding the smartphone briefly becomes less stable due to this form of activation. It might seem like nitpicking, but I believe there's an underlying discomfort. It just feels somehow unpleasant or outlandish to do it this way. A double tap is typically used in Android on larger spaces, like the wallpaper on the desktop, to, for example, turn off the screen. That's okay, since precision is secondary with such a large surface area. However, I'm not aware of instances where such action is used for small areas in Android app navigation. Holding down, on the other hand, is much more familiar. Think about bringing up the context menu by holding down an app, for example. That's something we know, and it just feels right. Additionally, implementing the activation of the context menu with a small haptic feedback could further enhance satisfaction. There's another point about the current implementation that I need to address because it's negatively noticeable: due to the current double tapping, there's inevitably a short delay when pressing the search button once. Sure, it's necessary due to the nature of the current double tapping feature, but I also don't think it's a good user experience. It feels like the button is responding sluggishly. Even though it's very brief, you miss that certain snappiness. It doesn't feel good, especially since you constantly use the button. And all this just because the current navigation prioritizes a feature that is typically only used once and then never again. Then there's the inconsistency of the visibility of the floating button. In my opinion, a floating button should be available everywhere and at all times in an app. Here, I refer to the implementation in the Google Drive app. In my opinion, a floating button is distinguished precisely by floating above everything and thus not being influenced by the app's activities. However, I think this inconsistency is likely due to the button's current temporary implementation.

 Anyway, I just had to get that off my chest. I think I've been able to justify my criticism well. If there are any further questions or comments, feel free to ask. Let it all out.

---

_All important points briefly summarized by ChatGPT_

The new floating search button is a great addition but has implementation issues: 

• Its movable nature seems unnecessary and risks accidental repositioning.

• Suggest moving the button's customization to settings to avoid accidental changes.

• Activating settings through a long press instead of double tap would improve user experience.

• Double tapping feels unfamiliar for small areas in app navigation; long press is more intuitive.

• Implementing haptic feedback for long press activation could enhance user satisfaction.

• The slight delay caused by double tapping diminishes button responsiveness.

• The floating button's visibility should be consistent throughout the app like in Google Drive.

Overall, while appreciating the update, there are usability and consistency concerns that need addressing.

**Comment by NikunjK8** [comment]
Ok my takes:-
I'm actually fine with hold to move and the short delay for search bar to appear. I'm not a developer but I'm guessing moving button customization to settings will take too much work to implement because it have to open a seperate window from settings. Fast4x is probably already tired of all the complains, it's a passion project afterall. Again not saying you're wrong, all your worries are legitimate but for me I'm fine with what it is now. I don't need this app to be perfect in every area. If i have to choose between user experience and features, i would choose features (As long as user experience is at least good enough which is the case for RiMusic).

**Reply by dAtA-TRoN** [reply]
I find your concerns about the effort quite reasonable, but I believe it's not as demanding as you think. Judging by how I know fast4x, he can easily handle it. And he's definitely capable of saying no to anything suggested here. He has turned down things from me many times, so I know he's capable. Always remember: these are just suggestions, not commands ☺️.

**Reply by NikunjK8** [reply]
Yeah but too much work, even your favorite work can burn you out after some time. Sometimes kind hearted people aren't as vocal about their opinions as they want to be. Either way I'm learning programming. Hopefully in few months if i learn Kotlin, I'll be able to help him for small tasks at least.

**Comment by NikunjK8** [comment]
Personally i still think having a seperate button for quick actions will solve all the problems and leave search button alone just like other apps. As [mentioned here](https://github.com/fast4x/RiMusic/issues/1419#issuecomment-2042067964). OR dropping the idea of a multi-purpose button all together and moving settings to top, again just like other apps.

**Reply by dAtA-TRoN** [reply]
Maybe. I don't know. Let's wait and see what fast4x has to say about it.

**Comment by NikunjK8** [comment]
Or moving settings and possibly other options behind this. This area isn't functional now and in side navigation bar it doesn't really matter because there's already enough space to add buttons. I'll only matter for bottom & top navs.

![IMG_20240413_105808](https://github.com/fast4x/RiMusic/assets/99175083/67b99fcd-41ad-492b-912e-e56a47a47ffd)

**Reply by dAtA-TRoN** [reply]
I think that's not such a good idea. The app logo shouldn't incorporate functionalities. Simply because it's not something people are accustomed to 🙃.

**Reply by NikunjK8** [reply]
Yeah fair point, I'm trying to come out with random ideas.

**Comment by sayyid5416** [comment]
Should've opened an issue instead of discussion. Easier to track that way

**Reply by dAtA-TRoN** [reply]
Perhaps. But I did it this way because I actually want to discuss this here more. It has a less instructional character. I didn't want to come in with guns blazing right away.

**Comment by sayyid5416** [comment]
btw I agree with all your points (the ones summarised by chatGPT)

**Reply by dAtA-TRoN** [reply]
Well, now we should wait and see what fast4x has to say about it.

**Comment by fast4x** [comment]
It seems to me a good compromise to highlight the settings. This would allow you to move the floating icon with long pressure and keep searching with single click

![Screenshot_20240413_183433](https://github.com/fast4x/RiMusic/assets/1484476/0909d6dd-4b63-442f-b559-a999785eb68f)

**Reply by NikunjK8** [reply]
Keep in mind there's already people talking about making these button optional on main screen and ability to add them in navigation bar because it doesn't look good enough when side bar is enabled.
#1488

**Reply by fast4x** [reply]
Just implemented

**Reply by sayyid5416** [reply]
btw, **why not** fix it at a suitable place instead of having the ability of changing its position?

**Reply by dAtA-TRoN** [reply]
![proxy-image(1).gif](https://github.com/fast4x/RiMusic/assets/71184627/9c7e3539-b20f-4372-8aeb-15027f8566af)

I'm still unclear about why the button needs to be movable constantly. What's the benefit of this functionality? I'm struggling to envision any scenario where such a feature would be necessary 🤯!

**Reply by fast4x** [reply]
> btw, **why not** fix it at a suitable place instead of having the ability of changing its position?

Make it fixed in a position that is, you don’t read and click on something that gets under it. It is more correct that if it happens that the button makes it impossible to read or click, you move it...

**Reply by fast4x** [reply]
> ![proxy-image(1).gif](https://github.com/fast4x/RiMusic/assets/71184627/9c7e3539-b20f-4372-8aeb-15027f8566af)
> 
> I'm still unclear about why the button needs to be movable constantly. What's the benefit of this functionality? I'm struggling to envision any scenario where such a feature would be necessary 🤯!

Make it fixed in a position that is, you don’t read and click on something that gets under it. It is more correct that if it happens that the button makes it impossible to read or click, you move it...

**Reply by dAtA-TRoN** [reply]
Yes, I've just read your response about this in another thread. While what you're saying makes sense, it's also a peculiar solution in my opinion. And it's probably because the app wasn't initially programmed with such a button in mind. Well, it is what it is. I think I'd prefer to deactivate the button again in the future, if possible. I think it was more robustly handled before.

**Reply by fast4x** [reply]
What do you think about this concept? #1513

I like it...

**Reply by dAtA-TRoN** [reply]
> What do you think about this concept? #1513
> 
> I like it... 

I actually think this is the best solution yet. However, I don't feel that the search function necessarily needs to be included there. It should at least be possible to move it downwards optionally, perhaps even just into the bottom navigation bar. That would suit me fine. But yes, overall, I'm quite impressed with the idea.

**Comment by fast4x** [comment]
This means that the settings icon can also go in the menu

**Comment by fast4x** [comment]
![Screenshot_20240413_190517](https://github.com/fast4x/RiMusic/assets/1484476/4a89ae7c-7ae3-4636-b0fc-763997239aa3)

**Comment by fast4x** [comment]
![Screenshot_20240413_190315](https://github.com/fast4x/RiMusic/assets/1484476/c7f03134-127b-49f9-be19-8a628b76faf7)

**Comment by NikunjK8** [comment]
@fast4x Ok if a second floating button isn't possible then here's my solutions:-
1. Remove floating search bar all together. We have the option to add extra search button in navigation bar, make it permanent as a replacement. Yes it'll make nav bar a little cluttered when icon texts are enabled (you can make them take two lines later #1399) but we have the option to disable text as a workaround right now and it looks fine without texts and only icons. Does both of these sounds like good compromise? Also the position of search in bottom nav bar and floating search bar is close so people will get used to it more than other changes. For people with top or side navigation bars, well they're already comfortable with their buttons on those position so they're not going to miss floating button as much.
2. Now replace floating search button with the new action button which you really liked as you [mentioned here.](https://github.com/fast4x/RiMusic/issues/1419#issuecomment-2041350557) Move settings behind this button and a lot of features could be added behind this in future. People will not struggle to find settings as much as they're doing now because new button with new icon will automatically catch the curiosity of everyone. Make it single click to reveal all the options behind it. Settings in bottom or top nav bar really doesn't make sense. It takes extra space and it's not a commonly used button unlike other buttons in nav bar. It only looks good in side nav bar.
 
I've tried to find a balance for everyone's opinion. What do you think?

**Reply by sayyid5416** [reply]
It's hard for people to find settings because no one expects to double click on action button. It's either one click or hold.

**Reply by NikunjK8** [reply]
One click was reserved for search & Hold for moving so there wasn't any other way to implement it anyway.
With a multi-purpose action button, a single click will bring out the settings icon behind it.

**Reply by dAtA-TRoN** [reply]
Right after the update, I think everyone initially tried to long-press the new floating button, assuming that's how you access the settings (except those diligent souls who read the changelog beforehand 🙄).

**Reply by NikunjK8** [reply]
Good thing i read the changelog lol. That prevent me from making unnecessary issues as well.

**Reply by OrangeZXZ** [reply]
I was inactive for some time and saw a big search icon at the bottom I find this really bad but gived me so mutch ideas for upgrades im going to start right now an issue with ideas

**Reply by dAtA-TRoN** [reply]
> Good thing i read the changelog lol. That prevent me from making unnecessary issues as well.

I see it differently: if the feature had been implemented more intuitively (press & hold for example 😃), nobody would have opened 'unnecessary issues', wouldn't have needed to read the changelog, and still would have found the settings 🙃.

**Reply by dAtA-TRoN** [reply]
> I was inactive for some time and saw a big search icon at the bottom I find this really bad but gived me so mutch ideas for upgrades im going to start right now an issue with ideas

I'm curious what you come up with.

**Reply by NikunjK8** [reply]
>I see it differently: if the feature had been implemented more intuitively (press & hold for example 😃), nobody would have opened 'unnecessary issues', wouldn't have needed to read the changelog, and still would have found the settings 🙃. 

Well true but again hold was reserved so it made sense from fast4x pov to implement it that way. The idea of multi purpose button is addressing it.

**Reply by sayyid5416** [reply]
I still don't understand why moving it freely is required.
- Click to open search
- Hold to open settings and others menu
- Give a setting to move action bar to left or right _(just like navigation bar)_

**Reply by NikunjK8** [reply]
For people who prefer different positions.

**Reply by sayyid5416** [reply]
what other positions could they use! It's not like they are gonna put it in the middle of the page.

**Reply by dAtA-TRoN** [reply]
> what other positions could they use! It's not like they are gonna put it in the middle of the page.

Read this. fast4x has just given the reason why the button must be movable:

![Screenshot_2024-04-13-21-28-15-73_320a9a695de7cdce83ed5281148d6f19.jpg](https://github.com/fast4x/RiMusic/assets/71184627/d09660c2-6a5a-49fc-8c14-587a39cdb363)

**Reply by sayyid5416** [reply]
Now it make sense lol 😅

**Reply by dAtA-TRoN** [reply]
> Now it make sense lol 😅

Yes, I think I will deactivate this floating button in the future, if possible. I found it better resolved before.

**Reply by sayyid5416** [reply]
We don't have to go that harsh. This Action Menu idea is awesome, we just need to refine it.

**Reply by fast4x** [reply]
I make it enable/disable by settings

**Reply by OrangeZXZ** [reply]
I made a new issue with 3 ideas one being that the floating icon could be an accessibility option

**Reply by Munsterguy** [reply]
Hi all. Sorry to bother ye. I but I have Rimusic on my Car and phone and TV. But with the new update I can't get into the settings anymore. I think your Rimusic app is brilliant. Please keep away from Google. I am only a person who loves Music and with yer app I can do that every day with out Hitler telling what to do. If iam in the wrong place. Please point me where to go.

**Reply by OrangeZXZ** [reply]
> Hi all. Sorry to bother ye. I but I have Rimusic on my Car and phone and TV. But with the new update I can't get into the settings anymore. I think your Rimusic app is brilliant. Please keep away from Google. I am only a person who loves Music and with yer app I can do that every day with out Hitler telling what to do. If iam in the wrong place. Please point me where to go. 

dont worry this feature will be removed next update

**Reply by Munsterguy** [reply]
Thank you OrangeZXZ

**Reply by fast4x** [reply]
> Hi all. Sorry to bother ye. I but I have Rimusic on my Car and phone and TV. But with the new update I can't get into the settings anymore. I think your Rimusic app is brilliant. Please keep away from Google. I am only a person who loves Music and with yer app I can do that every day with out Hitler telling what to do. If iam in the wrong place. Please point me where to go. 

Thanks, with next version try it and send feedback if will be ok

**Reply by Munsterguy** [reply]
Would be my Pleasure.

**Reply by Munsterguy** [reply]
Thanks for the update. Great work again

**Reply by fast4x** [reply]
> Thanks for the update. Great work again

You are welcome, next version report again if a trouble occur.

**Reply by Munsterguy** [reply]
Will Do.

**Reply by Munsterguy** [reply]
Hi ALL.
The Close app with the back button is not working on my Car radio but it works on my phone.
2nd Equalizer, Interact with the system Equalizer, when I press it, it says couldn't find an application to Equalize audio.
Can you please help with these issues? I do not know if these happen to other people or if is it just mine. your help is much appreciated and have a great weekend.

**Reply by OrangeZXZ** [reply]
> Hi ALL.
> The Close app with the back button is not working on my Car radio but it works on my phone.
> 2nd Equalizer, Interact with the system Equalizer, when I press it, it says couldn't find an application to Equalize audio.
> Can you please help with these issues? I do not know if these happen to other people or if is it just mine. your help is much appreciated and have a great weekend.

you can make an issue that way it could be added in the future

**Reply by Munsterguy** [reply]
Sorry QrangeZXZ, I am not great with tech and computers. What  you and your colleagues Do in a second would take me a lifetime Lol,
you would need to guide me like you would a blind man across a street

**Reply by OrangeZXZ** [reply]
> Sorry QrangeZXZ, I am not great with tech and computers. What  you and your colleagues Do in a second would take me a lifetime Lol,
> you would need to guide me like you would a blind man across a street

Don't worry i can do it for you

**Comment by fast4x** [comment]
Added setting to enable also in RiMusic ui, default is disabled.

**Reply by sayyid5416** [reply]
Thanks for making it disabled by default

**Comment by Munsterguy** [comment]
Hi lads. Great work on Rimusic. Can I ask when I go to ad ceelo Green. Song fcuk you. It says cant play server restrictions. And may I ask is there away I can transfer my playlist that I have on my phone To my my Car radio and TV . Instead of adding the playlist one by one. It's so time consuming. As I said before iam not tech savvy. Thanks again for all yer work. Be safe

**Reply by fast4x** [reply]
Sorry i don't understand exactly

**Reply by NikunjK8** [reply]
This song isn't playing for him, showing server restriction:- https://music.youtube.com/watch?v=-6srPIwiNCA

**Reply by Munsterguy** [reply]
NikunjKhangwal. Thanks I can play the song  on youtube but it won't download or play on Rimusic app. And can someone please help with downloading my music that I have downloaded on to my fone so I can transfer to my car and TV. as I love to hear new songs and download them on to Rimusic every day

**Reply by fast4x** [reply]
RiMusic play from api, if this is restricted, i don't to do anything

**Reply by NikunjK8** [reply]
Rimusic doesn't have any kind of sync or auto backup functionality right now. So you cannot replicate your data between devices.

**Reply by fast4x** [reply]
It can backup from one device and restore in other device...

**Reply by Munsterguy** [reply]
Thanks Lads I appreciate ye taking the time to answer me. Fast4x. How would I backup from one device and restore to another device. Please remember I am not tech savvy.

**Reply by NikunjK8** [reply]
From settings

![IMG_20240519_011532](https://github.com/fast4x/RiMusic/assets/99175083/e58e2f86-da44-4d40-9c8e-d8c8db7a6067)

**Reply by Munsterguy** [reply]
Do I put it on to a USB NikunjKhangwal and transfer to another device . As I say I am not tech savvy. Much help is truly appricated

**Reply by NikunjK8** [reply]
You can use anything you like for file transfer, everything works. Apps, usb, cable etc, whatever you were using to transfer data between your devices already. "Save to backup" will create a backup file in any folder you want and "restore from backup" will restore the playlists.
First save a backup file from the device with most up to date playlists, then share it to other devices and restore them.

**Reply by Munsterguy** [reply]
I will give a go. Thank you very much NikunjKhangwal your a Gentleman and a scholar.

**Comment by Munsterguy** [comment]
Hi lads just downloaded new update on Rimusic. Every time I go to play a song from my playlist. I Rimusic shuts down. Have a go one

**Reply by NikunjK8** [reply]
Working fine here. Let see if someone else also faces the issue so it can be identified

**Reply by fast4x** [reply]
Too generic info, try add some details please.

**Reply by OrangeZXZ** [reply]
> Hi lads just downloaded new update on Rimusic. Every time I go to play a song from my playlist. I Rimusic shuts down. Have a go one

Yes playing any song just closes the app and give me the option to clear cache (android pop-up)

**Reply by NikunjK8** [reply]
Hopefully someone can share crash log. Use [LogFox](https://github.com/F0x1d/LogFox) app to collect it. I would've done it but I'm not facing the issue so far.

**Reply by fast4x** [reply]
Change any background from cover...

**Comment by Munsterguy** [comment]


**Reply by NikunjK8** [reply]
These are crash logs. If you can copy them, share the entire list here so fast4x can get some help.

**Reply by fast4x** [reply]
change background options in appearance settings 

set background from theme and restart app
![Screenshot_2024-05-24-19-47-46-682_it.fast4x.rimusic.jpg](https://github.com/fast4x/RiMusic/assets/1484476/7f70925c-f4c4-447b-a74b-3eba11f30dd0)

**Reply by fast4x** [reply]
disable material you theme if enabled

**Reply by NikunjK8** [reply]
Tried with "background from theme", also I'm using dynamic theme. Material you is already disabled. Everything still working fine

**Reply by fast4x** [reply]
ok so it's backgrounds from cover buggy I'll fix it soon

**Comment by Munsterguy** [comment]


**Reply by Munsterguy** [reply]
hope this helps

**Reply by fast4x** [reply]
click on View summary

**Reply by Munsterguy** [reply]
Fast4x. Go back over text I send in here to see view summary.

**Reply by fast4x** [reply]
read #1496 i wrote

**Comment by Munsterguy** [comment]
Hi lads.  I am not technically trained. I never went to college. I left school early. Iam just an ordinary Joe soap who likes using Rimusic. I can only give what I see in front of me. When ye talk I don't understand yer lingo. I just download music and play it.

**Reply by dAtA-TRoN** [reply]
> I just download music and play it.

That's all you had to do. Welcome. If you come up with interesting suggestions, always put them out there.

**Reply by Munsterguy** [reply]
Hi Lads. Everyone involved with Rimusic. I downloaded and updated my Rimusic and it works. All I can say is Thank you very much for the hard work ye do. I like to listen to music instead of news about the world. All you here is bad news. But when I listen to my playlist it mahes me happy 😊 and I ask again keep it away from Google play. Ye are Legends.

**Reply by fast4x** [reply]
Thanks a lot

**Comment by Munsterguy** [comment]
Hi lads. How would you download Rimusic to windows 11 computer.

**Reply by NikunjK8** [reply]
It's a Android only app unfortunately. You have to use other options for windows.

**Reply by fast4x** [reply]
Try this https://www.androidauthority.com/how-to-run-android-apps-on-windows-11-3048569/

In the future is planned RiMusic as  multiplatform app, but not for iOS.

**Reply by OrangeZXZ** [reply]
> Try this https://www.androidauthority.com/how-to-run-android-apps-on-windows-11-3048569/
> 
> In the future is planned RiMusic as  multiplatform app, but not for iOS.

Technically WSA works but it was stopped (that means android 13max) we can download with magisk on github and I have it

**Reply by fast4x** [reply]
> Try this https://www.androidauthority.com/how-to-run-android-apps-on-windows-11-3048569/
> 
> In the future is planned RiMusic as  multiplatform app, but not for iOS.

or try Bluestacks solution 
https://www.bluestacks.com/bluestacks-5.html

**Comment by Munsterguy** [comment]
Thanks Lads. Blue stacks worked

**Reply by fast4x** [reply]
Make a short video and upload it here? Can you show us how it looks on Windows RiMusic?
